File: Project_Euler/JavaScript/Problem3.js
function isPrime(n) {
    if (n % 2 === 0) {
        return false;
    }
    var sqrtn = Math.sqrt(n);
    for (var i = 3; i < sqrtn + 1;) {
        if (n % i === 0) {
            return false;
        }
        i += 2;
    }
    return true;
}

var sqrtNum = Math.sqrt(600851475143);
var out = 0;
for (var i = 3; i < sqrtNum; i++) {
    if (600851475143 % i === 0 && isPrime(i)) {
        out = i;
    }
}
console.log(out);

File: Project_Euler/Python/Problem3.py
import numpy as np

def isPrime(number):
    for i in np.arange(3.0,np.sqrt(number),2):
        if number % i == 0:
            return 0
    return 1
def lpf(n):
    for i in np.arange(1.0,np.sqrt(n),2):
        if n % i == 0 and isPrime(i):
            prime = i
    return prime

print(lpf(600851475143))
